components/form/fields.tsx

```diff
@@ -0,0 +1,63 @@
+import { useId } from "@radix-ui/react-id";
+import { forwardRef, ReactNode } from "react";
+import { FormProvider, UseFormReturn } from "react-hook-form";
+
+import classNames from "@lib/classNames";
+
+export const Input = forwardRef<HTMLInputElement, JSX.IntrinsicElements["input"]>(function Input(props, ref) {
+  return (
+    <input
+      {...props}
+      ref={ref}
+      className={classNames(
+        "mt-1 block w-full border border-gray-300 rounded-sm shadow-sm py-2 px-3 focus:outline-none focus:ring-neutral-500 focus:border-neutral-500 sm:text-sm",
+        props.className
+      )}
+    />
+  );
+});
+
+export function Label(props: JSX.IntrinsicElements["label"]) {
+  return (
+    <label {...props} className={classNames("block text-sm font-medium text-gray-700", props.className)}>
+      {props.children}
+    </label>
+  );
+}
+
+export const TextField = forwardRef<
+  HTMLInputElement,
+  {
+    label: ReactNode;
+  } & React.ComponentProps<typeof Input> & {
+      labelProps?: React.ComponentProps<typeof Label>;
+    }
+>(function TextField(props, ref) {
+  const id = useId();
+  const { label, ...passThroughToInput } = props;
+
+  // TODO: use `useForm()` from RHF and get error state here too!
```

Comment:
Planning for these components to automatically handle error state from react hook form